msal/src/main/java/com/microsoft/identity/client/internal/controllers/BrokerMsalController.java

```diff
@@ -0,0 +1,184 @@
+import com.microsoft.identity.common.internal.request.AcquireTokenSilentOperationParameters;
+import com.microsoft.identity.common.internal.request.MsalBrokerRequestAdapter;
+import com.microsoft.identity.common.internal.result.AcquireTokenResult;
+import com.microsoft.identity.common.internal.result.MsalBrokerResultAdapter;
+
+/**
+ * The implementation of MSAL Controller for Broker
+ */
+public class BrokerMsalController extends BaseController {
+
+    private static final String TAG = BrokerMsalController.class.getSimpleName();
+
+    private BrokerResultFuture mBrokerResultFuture;
+
+    @Override
+    public AcquireTokenResult acquireToken(AcquireTokenOperationParameters parameters)
+            throws  InterruptedException, BaseException {
+
+        //Create BrokerResultFuture to block on response from the broker... response will be return as an activity result
+        //BrokerActivity will receive the result and ask the API dispatcher to complete the request
+        //In completeAcquireToken below we will set the result on the future and unblock the flow.
+        mBrokerResultFuture = new BrokerResultFuture();
+
+        //Get the broker interactive parameters intent
+        final Intent interactiveRequestIntent = getBrokerAuthorizationIntent(parameters);
+
+        final MsalBrokerRequestAdapter msalBrokerRequestAdapter = new MsalBrokerRequestAdapter();
+        interactiveRequestIntent.putExtra(
+                AuthenticationConstants.Broker.BROKER_REQUEST_V2,
+                msalBrokerRequestAdapter.brokerRequestFromAcquireTokenParameters(parameters)
+        );
+
+        //Pass this intent to the BrokerActivity which will be used to start this activity
+        final Intent brokerActivityIntent = new Intent(parameters.getAppContext(), BrokerActivity.class);
+        brokerActivityIntent.putExtra(BrokerActivity.BROKER_INTENT, interactiveRequestIntent);
+
+        //Start the BrokerActivity
+        parameters.getActivity().startActivity(brokerActivityIntent);
+
+        //Wait to be notified of the result being returned... we could add a timeout here if we want to
+        final Bundle resultBundle = mBrokerResultFuture.get();
+
+        return getAcquireTokenResult(resultBundle);
+
+    }
+
+    /**
+     * Get the intent for the broker interactive request
+     *
+     * @param request
+     * @return
+     */
+    private Intent getBrokerAuthorizationIntent(@NonNull final AcquireTokenOperationParameters request) throws ClientException {
+        Intent interactiveRequestIntent;
+        IMicrosoftAuthService service;
+
+        final MicrosoftAuthClient client = new MicrosoftAuthClient(request.getAppContext());
+        final MicrosoftAuthServiceFuture authServiceFuture = client.connect();
+
+        try {
+            service = authServiceFuture.get();
+            interactiveRequestIntent = service.getIntentForInteractiveRequest();
+
+            // TODO : See what's the right exception here
```

Comment:
My assumption would be that these errors roll-up to `ClientException` unless a good case can be made for any other error

Reply:
Agreed, missed to resolve todo's , fixed them